src/choreo/ghost/fd_ghost.h

```diff
@@ -35,12 +35,52 @@
 
    In-memory representation:
 
-   - Each tree ele is keyed by slot number.
+   There are two maps, both backed by the same pool of tree elements.
+
+   - One map is keyed by slot number, and one map is keyed by block_id (bid).
+
+   - The elements in the slot map are a subset of the elements in the
+     bid map.
 
    - Each tree ele tracks the amount of stake (`stake`) that has voted
      for its slot, as well as the recursive sum of stake for the subtree
      rooted at that ele (`weight`).
 
+   The map keyed by slot is the "happy tree." i.e. the first version of
```

Comment:
nice docs -- minor nits is double space after sentence period and maybe make three copies of diagram for the two versions of the tree if you only saw one version of the duplicate